unit/src/main/include/robot/state.h

```diff
@@ -20,6 +21,12 @@ class State {
     std::vector<std::vector<cv::Point2f>> detectedArucoCorners, rejectedArucoCorners;
     std::vector<int> detectedArucoIds;
     std::vector<cv::Point2f> transform_dst;
+
+    //Uploading
+    long lastTimeS;
+    bool doUpload = false;
+    bool uploadValues[constants::general::kNumMeasurements] = { true, false, true, false, true, false, true, false };
+
```

Comment:
I will give you a list of variables to "track" and I want them to be uploaded accordingly - all of that is not a state thing (state has the individual values), it is fully something that should stay within the subsystem. I might be misinterpreting things though

Reply:
The list of variables to track is something that is fully configured within the calculate method within the subsystem itself